pkg/providers/vsphere/vsphere.go

```diff
@@ -915,6 +915,54 @@ func (p *vsphereProvider) SetupAndValidateUpgradeCluster(ctx context.Context, cl
 	if err != nil {
 		return fmt.Errorf("failed setup and validations: %v", err)
 	}
+	err = p.validateMachineConfigsNameUniqueness(ctx, cluster, clusterSpec)
+	if err != nil {
+		return fmt.Errorf("failed validate machineconfig uniqueness: %v", err)
+	}
+	return nil
+}
+
+func (p *vsphereProvider) validateMachineConfigsNameUniqueness(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error {
+	prevSpec, err := p.providerKubectlClient.GetEksaCluster(ctx, cluster, clusterSpec.GetName())
+	if err != nil {
+		return err
+	}
+
+	cpmc := clusterSpec.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
+	if prevSpec.Spec.ControlPlaneConfiguration.MachineGroupRef.Name != cpmc {
+		em, err := p.providerKubectlClient.SearchVsphereMachineConfig(ctx, cpmc, cluster.KubeconfigFile, clusterSpec.GetNamespace())
+		if err != nil {
+			return err
+		}
+		if len(em) > 0 {
```

Comment:
Just for my understanding, why don't we allow to reference a `VsphereMachineConfig` that already exists? I'm assuming that if it already exits it's because it was created by being referenced by a different eks-a cluster

A `VsphereMachineConfig` is just a "template` for a VM configuration, why couldn't it be shared by multiple clusters?

Reply:
@mdsgabriel and i chatted offline. for the first path we dont want to allow sharing machineconfigs, especially btw management and workload cluster. e.g. if user upgrade workload cluster with shared machine config like network,  all machines in the management will have to be recreated. 
also the current delete workload cluster workflow deletes all eksa resources associated with it. If we allow sharing, it will also remove management's resources.

we can iterate after this release to be less restricted. but for now we add this validation to prevent unexpected behaviors

Reply:
> @mdsgabriel and i chatted offline. for the first path we dont want to allow sharing machineconfigs, especially btw management and workload cluster. e.g. if user upgrade workload cluster with shared machine config like network, all machines in the management will have to be recreated. also the current delete workload cluster workflow deletes all eksa resources associated with it. If we allow sharing, it will also remove management's resources.
> 
> we can iterate after this release to be less restricted. but for now we add this validation to prevent unexpected behaviors

yes. we can change the in the future